addons/edi/models/edi_synchronizer.py

```diff
@@ -317,12 +333,18 @@ def execute(self):
         # Process records in order of lookup relationship readiness
         remaining = self
         offset = 0
+        clear_cache_count = 0
         while remaining:
 
             # Identify records for which all lookup relationships are ready
             ready = remaining._add_edi_relates(required=False)
-            if remaining and not ready:
-                remaining._add_edi_relates(required=True)
+            if not ready:
+                ready = remaining._add_edi_relates(required=True)
+            if not ready:
```

Comment:
can this be indented? it reads a bit odd but it's only ever relevant inside the above if

Reply:
Can you make an example? I'm struggling to understand which way you would like me to change it.

Reply:
```python
if not ready:
    ready = remaining._add_edi_relates(required=True)
    if not ready:
        # No more progress can be made. Do not raise an error, since
        # fail_fast must be off here, otherwise
        # _add_edi_relates(required=True) would have already raised one.
        break
```
like this
because it's only relevant in this branch, it reads odd to say
```python
if not ready:
    ready  = ...
if not ready:
    ...
```

Reply:
@PeterAlabaster Addressed.